modules/lang-painless/src/main/java/org/elasticsearch/painless/node/ELambda.java

```diff
@@ -162,8 +162,8 @@ void analyze(Locals locals) {
             }
         }
         // prepend capture list to lambda's arguments
-        List<String> paramTypes = new ArrayList<>();
-        List<String> paramNames = new ArrayList<>();
+        List<String> paramTypes = new ArrayList<>(captures.size());
+        List<String> paramNames = new ArrayList<>(captures.size());
```

Comment:
This one is not right, it should be `captures.size() + actualParamTypeStrs.size()` and `captures.size() + paramNameStrs.size()`.